demophin.py

```diff
@@ -1,7 +1,21 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
 
 import sys
 import os
 import json
+import logging
+
+# LTA 2016-05-12
+import importlib
+
+def preprocess(sent, processor_name):
+    try:
+        i = importlib.import_module(processor_name)
+        return i.get_ace_input(sent)
+    except Exception as e:
+        logging.error("Preprocessor cannot be used (Error = %s)" % (e,))
+        pass
```

Comment:
I imagine that some preprocessors might not be Python modules, in which case they would need Python wrappers. What about using the `subprocess` module and piping the sentence through an executable filter? This would work for Python modules, shell scripts, binary applications, etc...

Otherwise, the catch-all `except` block hides what the error might have been. Did it fail to import? Did the module not have the `get_ace_input()` function? Did that function return an error?